addons/website_sale/controllers/product_configurator.py

```diff
@@ -1,147 +1,323 @@
+        :param datetime date: The date to compute the strikethrough price at.
+        :param float price: The actual price of the product.
+        :rtype: float|None
+        :return: The strikethrough price of the product, if there is one.
+        """
+        # Don't apply taxes to `compare_list_price`, it should be displayed as is.
+        compare_list_price = product_or_template.currency_id._convert(
+            from_amount=product_or_template.compare_list_price,
+            to_currency=currency,
+            company=request.env.company,
+            date=date,
+            round=False,
+        )
+        list_price = self._apply_taxes_to_price(
+            product_or_template._price_compute('list_price', currency=currency)[
+                product_or_template.id
+            ],
+            product_or_template,
+            currency
+        )
+
+        # First, try to use `compare_list_price` as the strikethrough price.
+        if (
+            request.env.user.has_group('website_sale.group_product_price_comparison')
+            and compare_list_price
+        ):
+            # If `compare_list_price` is lower than `price`, don't show it, but don't fall back on
+            # `list_price` either.
+            if currency.compare_amounts(compare_list_price, price) == 1:
+                return compare_list_price
+        # Second, try to use `list_price` as the strikethrough price.
+        elif (
+            pricelist.discount_policy == 'without_discount'
+            and currency.compare_amounts(list_price, price) == 1
+        ):
+            return list_price
+
+        return None
+
+    def _should_show_product(self, product_template, parent_combination):
+        """ Override of `sale` to only show products that can be added to the cart.
+
+        :param recordset product_template: The product being checked, as a `product.template`
+                                           record.
+        :param recordset parent_combination: The combination of the parent product, as a
+                                             `product.template.attribute.value` recordset.
+        :rtype: bool
+        :return: Whether the product should be shown in the configurator.
+        """
+        should_show_product = super()._should_show_product(product_template, parent_combination)
+        if request.is_frontend:
+            return (
+                should_show_product
+                and product_template._is_add_to_cart_possible(parent_combination)
+            )
+        return should_show_product
+
+    @staticmethod
+    def _populate_currency_and_pricelist(kwargs):
+        website = request.website
+        kwargs.update({
+            'currency_id': website.currency_id.id,
+            'pricelist_id': website.pricelist_id.id,
+        })
+
+    @staticmethod
+    def _apply_taxes_to_price(price, product_or_template, currency):
+        product_taxes = product_or_template.sudo().taxes_id.filtered(
```

Comment:
```suggestion
        product_taxes = product_or_template.sudo().taxes_id._filter_taxes_by_company(request.env.company)
```
new util to use (for subsidiary companies logic)